ui/app/templates/clients/client/index.hbs

```diff
@@ -835,5 +838,40 @@
         </div>
       </div>
     {{/if}}
+    {{#if (can "write client")}}
+      {{#if this.editingMetadata}}
+        <div class="add-dynamic-metadata">
+          <h3 class="title is-6">Add Dynamic Metadata</h3>
+          {{!-- TODO: add a cancel button to drop out of editingMetadata --}}
+          <MetadataEditor
+            @kv={{this.newMetaData}}
+            @onEdit={{this.validateKey}}
+          >
+            <button
+              disabled={{or (not this.newMetaData.key) (not this.newMetaData.value)}}
+              type="submit"
+              class="button is-primary"
+              {{on "click" (queue
+                (action this.addDynamicMetaData this.newMetaData)
```

Comment:
If this promise is rejected, the functions that come after this will not execute. We're not relying on the return value of any of these functions, just the promise resolution before we reset.

I like handling things in templates and I don't think you need to change this. But just documenting this, if we run into a problem where state doesn't update... we'll know why.